packages/components/src/site-thumbnail/use-mshots-img.tsx

```diff
@@ -96,7 +96,7 @@ export const useMshotsImg = (
 		}
 
 		return () => {
-			clearTimeout( timeout.current );
+			// clearTimeout( timeout.current );
```

Comment:
```bash
packages/components/src/site-thumbnail/use-mshots-img.tsx:99:18 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(timeoutId: Timeout): void', gave the following error.
    Argument of type 'Timeout | undefined' is not assignable to parameter of type 'Timeout'.
      Type 'undefined' is not assignable to type 'Timeout'.
  Overload 2 of 2, '(id?: number | undefined): void', gave the following error.
    Argument of type 'Timeout | undefined' is not assignable to parameter of type 'number | undefined'.
      Type 'Timeout' is not assignable to type 'number'.

99    clearTimeout( timeout.current );
```

Looks like we have to check `timeout.current` before calling `clearTimeout` to fix the following error. For example

```js
if ( timeout.current ) {
  clearTimeout( timeout.current );
}
```

Reply:
> Looks like we have to check `timeout.current` before calling `clearTimeout` to fix the following error.

Yes! Thanks for having a look. I will push that fix [separately](https://github.com/Automattic/wp-calypso/pull/71172). Because I wasn't able to produce the build issues locally, I wanted to push this basic change and see if the CI tests were happy or not.

> IMHO I thought @automattic/components should be basic components without any business logic including track events. It might be better to provider onClick prop to send the event from the parent component.

That makes a lot of sense to me. I'll look at pulling this apart in a follow-up PR.